web/src/main/java/org/fao/geonet/Geonetwork.java

```diff
@@ -558,6 +570,39 @@ private boolean checkDBWrite() {
         scheduledExecutorService.scheduleWithFixedDelay(DBHeartBeat, initialDelay, fixedDelay, TimeUnit.SECONDS);
     }
 
+    private void createLinkMonitor(final ResourceManager rm, final GeonetContext gc, String linkMonitorClass,
+                                   Integer initialDelay, Integer fixedDelay, Integer reindexInterval) {
+        logger.info("Creating Link Monitor...");
+
+        try {
+            Class classz = Class.forName(linkMonitorClass);
+            gc.linkMonitor = (LinkMonitorInterface) classz.newInstance();
+            gc.linkMonitor.init(rm, gc, logger, reindexInterval);
+            logger.info(String.format("Link Monitor with class '%s' initialized", linkMonitorClass));
+        } catch (Exception e) {
+            e.printStackTrace();
+            logger.error(String.format("Failed initializing Link Monitor with class '%s'", linkMonitorClass));
+            return;
+        }
+
+        ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(1);
+        Runnable LinkMonitorThread = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    gc.getLinkMonitor().check();
+                }
+                // any uncaught exception would cause the scheduled execution to silently stop
+                catch(Throwable x) {
+                    logger.error("Link Monitor error: " + x.getMessage() + " This error is ignored.");
```

Comment:
Like `DBHeartBeat` (my usual excuse). We can potentially abort execution here if the user does specify that he wants link monitoring enabled and an error occurred. That be good.